### PR TITLE
JAMES-2574 Listing Healthchecks

### PR DESCRIPTION
**Pull request by kratostaine**
JAMES-2574 Implemented route for listing health checks and added tests and documentation for Issue #1847

### CODE REVIEW

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/HealthCheckRoutesTest.java [hunk 1]
```diff
@@ -240,4 +238,38 @@ public void performHealthCheckShouldWorkWithEscapedPathParam() throws MalformedU
...
+           .statusCode(HttpStatus.OK_200);
+    }
+
+    @Test
+    public void getHealthchecksShouldReturnHealthCheckWhenHealthCheckPresent() {
+        healthChecks.add(healthCheck(Result.healthy(COMPONENT_NAME_3)));
+        when()
+           .get(HealthCheckRoutes.CHECKS)
+        .then()
+            .body("",hasSize(1))
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/HealthCheckRoutesTest.java]: You miss an extra space after this coma

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/HealthCheckRoutesTest.java [hunk 2]
```diff
@@ -240,4 +238,38 @@ public void performHealthCheckShouldWorkWithEscapedPathParam() throws MalformedU
...
+    }
+
+    @Test
+    public void getHealthchecksShouldReturnHealthChecksWhenHealthChecksPresent() {
+        healthChecks.add(healthCheck(Result.healthy(COMPONENT_NAME_2)));
+        healthChecks.add(healthCheck(Result.healthy(COMPONENT_NAME_3)));
+        when()
+           .get(HealthCheckRoutes.CHECKS)
+        .then()
+           .body("",hasSize(2))
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/HealthCheckRoutesTest.java]: Here also you miss a space after the coma

**Comment by kratostaine** [comment]
@chibenwa thank you so much for the quick response. Very much grateful for your guidance. I have pushed a commit with the changes.

File: server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/dto/HealthCheckDto.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.dto;
+
+import org.apache.james.core.healthcheck.ComponentName;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/dto/HealthCheckDto.java]: If you check the Jenkins CI report, you will see that the checkstyle is failing.

We do enforce a few cosmetic rules. You need to skip a line between org.* Imports and com.* Imports.

**Comment by chibenwa** [comment]
Thank you also very much for the quick fixes ;-)

File: server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/dto/HealthCheckDto.java (hunk 3 above)
Comment (kratostaine) [reply server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/dto/HealthCheckDto.java]: Thank you for the advise @chibenwa. I am not able to view the CI report and hence the trouble. I've added the linebreak.

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Hi,

This pull request has just been merged.

Thank you again for your dedication toward the James project. If you need help, want to [contribute](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20newbie%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC) [documentation](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20documentation%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), new [features](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20feature%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), tests or [bugfix](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20easyfix%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), please don't hesitate to reach us.

Thanks again,

Cheers,

Benoit

**Comment by kratostaine** [comment]
> Hi,
> 
> This pull request has just been merged.
> 
> Thank you again for your dedication toward the James project. If you need help, want to [contribute](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20newbie%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC) [documentation](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20documentation%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), new [features](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20feature%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), tests or [bugfix](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20easyfix%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), please don't hesitate to reach us.
> 
> Thanks again,
> 
> Cheers,
> 
> Benoit

Thank you so much for providing the opportunity to contribute. It's been a pleasure. Also thank you for the kind support throughout. Very much grateful for your help. It's my first contribution to open source and it feels great :)

Cheers
Madhu Bhat

**Comment by chibenwa** [comment]
@kratostaine I'm glad that the team allowed you to have such a pleasant experience. That had been a pleasure on our side too to work with someone motivated like you!

We often join "events" like:
  - [HacktoberFest](https://hacktoberfest.digitalocean.com/)
  - [24PullForChristmas](https://24pullrequests.com/)
  - [Google Summer of Code](https://summerofcode.withgoogle.com/)

We will also continue to propose some 'guided issues' on this repository. So that you can get the same kind of experience even when no event takes place.

Cheers,

Benoit Tellier

**Pull request closed without merging**